science/apbs/Portfile

```
@@ -20,64 +17,86 @@ long_description        APBS is a software package for the numerical \
+distfiles               v${version}.tar.gz 1.9.3.tar.gz 
+checksums               1.9.3.tar.gz \
+                        rmd160  25cf57754a0b817e76e19b8de93617e554302661 \
+                        sha256  2ce7ab04cba4403f4208c3ecf1c81a0a18aae6a77d22da0ffa5f64c2da7c6f28 \
+                        size    27757348 \
+                        v${version}.tar.gz \
+                        rmd160  17de250c3887ff68f89a873c5a0be19f6ab38813 \
+                        sha256  644e6246fd37c9dd4172fc7be1763337082ab8584dbcd53e738c831b62c89bb2 \
+                        size    85806314
+
+# Test support requires python3 but the
+# building the apbs python module with swig
+# is unnecessary for apbs's use in pymol.
+platform darwin {
+   if {${os.major} < 13} {
+       depends_test      port:python312
     }
 }
 
+depends_lib             port:maloc \
+                        port:superlu \
+                        port:SuiteSparse
+
+configure.optflags      -O3 -ffast-math -fcommon -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types 
+# Follow Debian's approach of not building those
+# features like BEM, GEOFLOW or PBAM that only
+# exist in apbs's github without tagged releases yet.
+configure.args-append   -DAPBS_STATIC_BUILD=ON \
+                        -DBUILD_SHARED_LIBS=OFF \
+                        -DBUILD_TOOLS=ON \
+                        -DENABLE_OPENMP:BOOL=OFF \
+                        -DENABLE_BEM=OFF \
+                        -DENABLE_GEOFLOW=OFF \
+                        -DENABLE_PBAM=OFF \
+                        -DBUILD_MALOC=OFF \
+                        -DBUILD_SUPERLU=OFF \
+                        -DBUILD_SHARED_LIBS=ON \
+                        -DFETCHCONTENT_SOURCE_DIR_FETK=${worksrcpath}/FETK-1.9.3 \
+                        -DCMAKE_C_COMPILER_ARG1:STRING="${configure.cflags} -I${worksrcpath}/include" \
+                        -DCMAKE_CXX_COMPILER_ARG1:STRING="${configure.cxxflags} -I${worksrcpath}/include -stdlib=libc++" \
+                        -DSUPERLU_INCLUDES=${prefix}/include -DSUPERLU_LIBRARIES=${prefix}/lib 
+
+patchfiles              cmake-maloc.patch disable_python_cmake_check.patch patch-testfiles.patch open-U-obsolete.patch python3.patch wrong-path-for-interpreter.patch
 test.run  yes
 test {
     ln ${worksrcpath}/tools/manip/inputgen.py ${worksrcpath}/tests/
     ln ${worksrcpath}/tools/manip/psize.py    ${worksrcpath}/tests/
 
-    if {[mpi_variant_isset]} {
-        reinplace "s|mpiexec|\"${mpi.exec}\", \"-n\", \"${build.jobs}\"|" ${worksrcpath}/tests/apbs_tester.py
+    platform darwin {
+       if {${os.major} < 13} {
+          system -W ${worksrcpath}/tests "PATH=${destroot}${prefix}/bin:/usr/bin:/bin:${prefix}/bin ${prefix}/bin/python3.12 apbs_tester.py -c test_cases.cfg || exit 0"
+       } else {
+          system -W ${worksrcpath}/tests "PATH=${destroot}${prefix}/bin:/usr/bin:/bin:${prefix}/bin python3 apbs_tester.py -c test_cases.cfg || exit 0"
+       }
```

This code duplication could be reduced. Also your indentation was not at multiples of 4 spaces. Also python3 is not available in the OS until macOS 10.15 (Darwin 19).
```suggestion
        system -W ${worksrcpath}/tests "PATH=${destroot}${prefix}/bin:/usr/bin:/bin:${prefix}/bin ${python} apbs_tester.py -c test_cases.cfg || exit 0"
```

---

Reduced duplicated code, fixed indentations and switched to python variable to handle python3 availability.